File: spec/python/test_cast_to_imported.py
import unittest

from cast_to_imported import CastToImported

class TestCastToImported(unittest.TestCase):
    def test_cast_to_imported(self):
        r = CastToImported.from_file("src/fixed_struct.bin")

        self.assertEqual(r.one.one, 0x50)
        self.assertEqual(r.one_casted.one, 0x50)

File: spec/lua/test_cast_to_imported.lua
local luaunit = require("luaunit")

require("cast_to_imported")

TestCastToImported = {}

function TestCastToImported:test_cast_to_imported()
    local r = CastToImported:from_file("src/fixed_struct.bin")

    luaunit.assertEquals(r.one.one, 0x50)
    luaunit.assertEquals(r.one_casted.one, 0x50)
end
